irohad/network/impl/block_loader_impl.cpp

```diff
@@ -23,39 +23,46 @@
 #include "builders/protobuf/transport_builder.hpp"
 #include "interfaces/common_objects/peer.hpp"
 #include "network/impl/grpc_channel_builder.hpp"
-#include "validators/default_validator.hpp"
 
 using namespace iroha::ametsuchi;
 using namespace iroha::network;
 using namespace shared_model::crypto;
 using namespace shared_model::interface;
 namespace val = shared_model::validation;
 
-BlockLoaderImpl::BlockLoaderImpl(std::shared_ptr<PeerQuery> peer_query,
-                                 std::shared_ptr<BlockQuery> block_query)
-    : peer_query_(std::move(peer_query)), block_query_(std::move(block_query)) {
-  log_ = logger::log("BlockLoaderImpl");
-}
+namespace {
+  const char *kPeerNotFound = "Cannot find peer";
+  const char *kTopBlockRetrieveFail = "Failed to retrieve top block";
+  const char *kPeerRetrieveFail = "Failed to retrieve peers";
+  const char *kPeerFindFail = "Failed to find requested peer";
+
+  /**
+   * If @param block_variant contains block, return it.
+   * If empty block return error
+   */
+  iroha::expected::Result<std::shared_ptr<Block>, std::string> getNonEmptyBlock(
```

Comment:
Is this function needed? As I see, later you unpack nonempty block with it and then immediately write a visitor again, but to unpack a result created by yourself. Maybe, it's better to match the block emptiness directly in the code, so you wouldn't have two visitors?

Reply:
I am not sure what do you mean by this comment. Block Factory returns result, and I would need to add visit_in_place inside match in order to retrieve block. Since the empty block is an error case in block loader, it is logical to apply this sort of transformation